core/schema/file.graphqls

```diff
@@ -28,6 +28,12 @@ type File {
     Location of the written directory (e.g., "output.txt").
     """
     path: String!
+
+    """
+    If allowParentDirPath is true, the path argument can be a directory path, in which case
+    the file will be created in that directory.
+    """
+    allowParentDirPath: Boolean
```

Comment:
cc @vito I added the ability for a file export path to just be a parent dir on the host, with the file name being set to whatever the file's name is in LLB.

I have some vague recollection of discussing this, but I don't remember the context at all. I just wanted to double check with you that it's not that there's some obscure case where this is a bad idea or something.

Reply:
Here's one dragon I found in the prior discussion: https://github.com/dagger/dagger/pull/3348#discussion_r1010942286

> Oh, yikes, the Buildkit exporter actually seems to do a os.RemoveAll of the destination directory in order to replace it with a file. So that actually "works" but doesn't do NEARLY what the user wanted. 😱

Is that still happening? 😄

Reply:
Oh right! I knew there was something weird around this but couldn't remember or find it, thank you for digging that up!

Interestingly, that isn't the behavior I'm seeing. I can export a single file into an existing directory and the rest of the existing contents are left alone. I'll poke around a bit to check there's not corner cases with undesirable behavior though.